File: Module/ALittleIDE/Other/GameLibrary/CEngine/JSScript/CEngine/MainTemplate/UIViewer.js
{
if (typeof ALittle === "undefined") ALittle = {};

ALittle.RegStruct(1225542858, "ALittle.UIViewModuleDebugInfo", {
name : "ALittle.UIViewModuleDebugInfo", ns_name : "ALittle", rl_name : "UIViewModuleDebugInfo", hash_code : 1225542858,
name_list : ["window_width","window_height","title","ui_name"],
type_list : ["int","int","string","string"],
option_map : {}
})

ALittle.g_Control = undefined;
ALittle.g_LayerGroup = undefined;
ALittle.g_BasePath = undefined;
ALittle.__Module_Setup = function(layer_group, control, base_path, debug) {
	ALittle.g_Control = control;
	ALittle.g_LayerGroup = layer_group;
	ALittle.g_BasePath = base_path;
	let [error, info] = (function() { try { let ___VALUE = lua.json.decode.call(undefined, debug); return [undefined, ___VALUE]; } catch (___ERROR) { return [___ERROR.message]; } })();
	let screen_width = ALittle.System_GetScreenWidth();
	let screen_height = ALittle.System_GetScreenHeight();
	let window_width = info.window_width;
	let window_height = info.window_height;
	let rate = 1.0;
	if (ALittle.System_GetPlatform() !== "Windows") {
		window_width = lua.math.floor(screen_width / screen_height * window_height);
	} else {
		if (window_width > screen_width || window_height > screen_height) {
			rate = 0.5;
		}
	}
	ALittle.System_CreateView(info.title, window_width, window_height, ALittle.UIEnumTypes.VIEW_RESIZABLE, rate);
	let layer = ALittle.NewObject(ALittle.DisplayLayout, ALittle.g_Control);
	layer.width_type = ALittle.UIEnumTypes.SIZE_MARGIN;
	layer.height_type = ALittle.UIEnumTypes.SIZE_MARGIN;
	ALittle.g_LayerGroup.AddChild(layer);
	ALittle.g_Control.CreateControl(info.ui_name, undefined, layer);
}

ALittle.PreLoadFile = function(ui_name, layer, result) {
}

ALittle.__Module_Shutdown = function() {
}

ALittle.__Module_GetInfo = function(control, base_path) {
	let info = {};
	info.title = "UIViewer";
	info.icon = undefined;
	info.width_type = ALittle.UIEnumTypes.SIZE_MARGIN;
	info.width_value = 0;
	info.height_type = ALittle.UIEnumTypes.SIZE_MARGIN;
	info.height_value = 0;
	return info;
}

ALittle.__Plugin_Setup = function(control, base_path, debug) {
}

ALittle.__Plugin_Shutdown = function() {
}

}

File: ALittle/Module/ALittleIDE/Other/GameLibrary/CEngine/Script/CEngine/MainTemplate/UIViewer.lua

module("ALittle", package.seeall)

local ___pairs = pairs
local ___ipairs = ipairs

RegStruct(1225542858, "ALittle.UIViewModuleDebugInfo", {
name = "ALittle.UIViewModuleDebugInfo", ns_name = "ALittle", rl_name = "UIViewModuleDebugInfo", hash_code = 1225542858,
name_list = {"window_width","window_height","title","ui_name"},
type_list = {"int","int","string","string"},
option_map = {}
})

g_Control = nil
g_LayerGroup = nil
g_ModuleBasePath = nil
function __Module_Setup(layer_group, control, module_base_path, script_base_path, debug)
	g_Control = control
	g_LayerGroup = layer_group
	g_ModuleBasePath = module_base_path
	local error, info = Lua.TCall(String_JsonDecode, debug)
	local screen_width = System_GetScreenWidth()
	local screen_height = System_GetScreenHeight()
	local window_width = info.window_width
	local window_height = info.window_height
	local rate = 1.0
	if System_GetPlatform() ~= "Windows" then
		window_width = Math_Floor(screen_width / screen_height * window_height)
	else
		if window_width > screen_width or window_height > screen_height then
			rate = 0.5
		end
	end
	System_CreateView(info.title, window_width, window_height, ALittle.UIEnumTypes.VIEW_RESIZABLE, rate)
	local layer = DisplayLayout(g_Control)
	layer.width_type = UIEnumTypes.SIZE_MARGIN
	layer.height_type = UIEnumTypes.SIZE_MARGIN
	g_LayerGroup:AddChild(layer)
	g_Control:CreateControl(info.ui_name, nil, layer)
end

function PreLoadFile(ui_name, layer, result)
end

function __Module_Shutdown()
end

function __Module_GetInfo(control, base_path)
	local info = {}
	info.title = "UIViewer"
	info.icon = nil
	info.width_type = UIEnumTypes.SIZE_MARGIN
	info.width_value = 0
	info.height_type = UIEnumTypes.SIZE_MARGIN
	info.height_value = 0
	return info
end

function __Plugin_Setup(control, module_base_path, script_base_path, debug)
end

function __Plugin_Shutdown()
end

